test/mod2pi.jl

```diff
@@ -194,3 +194,75 @@ testModPi()
 @test mod2pi(355.0f0) ≈ 3.1416228f0
 @test mod2pi(Int64(2)^60) == mod2pi(2.0^60)
 @test_throws ArgumentError mod2pi(Int64(2)^60-1)
+
+@testset "rem_pio2_kernel" begin
+    # test worst case
+    x = 6381956970095103.0 * 2.0^797
+    a = setprecision(BigFloat, 4096) do
+        rem(big(x), big(pi)/2, RoundNearest)
+    end
+
+    n, yrem = Base.Math.rem_pio2_kernel(x)
+    y=yrem.hi+yrem.lo
+    @test a-y<nextfloat(y)/2
+    # The following has easy and hard cases in each interval. A hard case is one
+    # where x ≈ k*pi/2 for some integer k.
+    cases = [0.0, pi/6, # -π/4 <= x <= π/4
+             2*pi/4-0.1, 2*pi/4, # -2π/4 <= x <= 2π/4
+             3*pi/4-0.1, # -3π/4 <= x <= 3π/4
+             pi-0.1, Float64(pi), # -4π/4 <= x <= 4π/4
+             5*pi/4-0.1, # -5π/4 <= x <= 5π/4
+             6*pi/4-0.1, 6*pi/4, # -6π/4 <= x <= 6π/4
+             7*pi/4-0.1, # -7π/4 <= x <= 7π/4
+             2*pi, 2*pi-0.1, # -8π/4 <= x <= 8π/4
+             9*pi/4-0.1, # -9π/4 <= x <= 9π/4
+             2.0^10*pi/4, # -2.0^20π/2 <= x <= 2.0^20π/2
+             2.0^30*pi/4, # |x| >= 2.0^20π/2, idx < 0
+             2.0^80*pi/4] # |x| >= 2.0^20π/2, idx > 0-0.22370138542135648
+
+     # ieee754_rem_pio2_return contains the returned value from the ieee754_rem_pio2
+     # function in openlibm https://github.com/JuliaLang/openlibm/blob/master/src/e_rem_pio2.c
```

Comment:
we may want to point to a release tag instead of master on this link in case this file changes or gets moved around in the future